components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx

```diff
@@ -336,32 +188,26 @@ const EditVisibleAssetsModal = ({ onClose }: Props) => {
         newList.splice(tokenIndex, 1, updatedToken)
         setUpdatedTokensList(newList)
       } else {
+        if (token.isErc721) setTokenContractAddress(token.contractAddress)
         setUpdatedTokensList(addOrRemoveTokenFromList(selected, token))
       }
     } else {
-      if (token.isErc721) {
-        setTokenContractAddress(token.contractAddress)
-        setShowTokenIDRequired(true)
-        setShowAddCustomToken(true)
-        return
-      }
       setUpdatedTokensList(addOrRemoveTokenFromList(selected, token))
     }
-  }
+  }, [isUserToken, updatedTokensList, addOrRemoveTokenFromList])
 
-  const toggleShowAddCustomToken = () => {
+  const toggleShowAddCustomToken = React.useCallback(() => {
     setShowAddCustomToken(!showAddCustomToken)
-  }
+  }, [showAddCustomToken])
 
-  const onClickSuggestAdd = () => {
+  const onClickSuggestAdd = React.useCallback(() => {
     setTokenContractAddress(searchValue)
     toggleShowAddCustomToken()
-  }
+  }, [searchValue, toggleShowAddCustomToken])
 
-  const onClickCancel = () => {
-    resetInputFields()
+  const onCancel = React.useCallback(() => {
```

Comment:
This method doesn't seem to be needed anymore, you can just call `toggleShowAddCustomToken`